### PR TITLE
Introducing OscarsStore

### PR DESCRIPTION
**Pull request by shtukas**

### CODE REVIEW

File: src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala [hunk 1]
```diff
@@ -61,4 +61,43 @@ class UserStore(client: AmazonDynamoDBAsyncClient, usersTableName: String, userT
 
   }
 
+  object OscarsStore {
+
+    def getUserNoms(id: String): Future[Option[UserNoms]] = {
+      val futureResult = ScanamoAsync.get[UserNoms](client)(userNomsTableName)('ID -> id)
+      futureResult.map { result =>
```
Comment (LATaylor-guardian) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: You could make use of a monad transformer here to reduce nesting I think. I'll let you read :)
https://enear.github.io/2016/04/11/monad-transformers/

Comment (shtukas) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: Thanks @LATaylor-guardian !

File: src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala [hunk 2]
```diff
@@ -61,4 +61,43 @@ class UserStore(client: AmazonDynamoDBAsyncClient, usersTableName: String, userT
...
+              None
+            }, {
+              Some(_)
+            }
+          )
+        }
+      }
+    }
+
+    def createUserNominationsRecordWithBestPicture(id: String, bestPicture: String): Future[PutItemResult] = {
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: I don't think this function is necessary, `update` will create the row if it's not already there

Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: actually, it should be a `put`, which will update or create:
http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html

File: src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala [hunk 3]
```diff
@@ -61,4 +61,43 @@ class UserStore(client: AmazonDynamoDBAsyncClient, usersTableName: String, userT
...
+      // The bestFilm variable is there because we are going to create the record only when it has been given by the user
+      ScanamoAsync.exec(client)(
+        userNomsTable.put(UserNoms(id, Some(bestPicture)))
+      )
+    }
+
+    def updateUserNominationsRecord(nominations: UserNoms): Unit = {
+      val table = Table[UserNoms](userNomsTableName)
+      table.update(
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: assuming you always retrieve the latest version of the dynamo record first, you can just do `put(nominations)`

File: src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala (hunk 1 above)
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: nice idea, `getUser` above could also benefit from some cats 🐱

Comment (shtukas) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: :cat:

File: src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala (hunk 2 above)
Comment (shtukas) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: Will do tomorrow. Thanks!

File: src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala (hunk 3 above)
Comment (shtukas) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: Ok, cool! Tomorrow :)

**Comment by shtukas** [comment]
@tomrf1 @LATaylor-guardian 
I have made all the corrections that we talked about. 

( @annebyrne  )

File: src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala [hunk 4]
```diff
@@ -61,4 +63,24 @@ class UserStore(client: AmazonDynamoDBAsyncClient, usersTableName: String, userT
 
   }
 
+  object OscarsStore {
+
+    def getUserNominations(id: String): Future[Option[UserNoms]] = {
+      val futureNominations = OptionT(ScanamoAsync.get[UserNoms](client)(userNomsTableName)('ID -> id))
+      futureNominations.map { result =>
+        result.fold({ error =>
+          appLogger.error(s"Error parsing User data from dynamodb: $error")
```
Comment (tomrf1) [inline src/main/scala/com/gu/facebook_news_bot/stores/UserStore.scala]: "Error parsing UserNoms data..."

**Comment by tomrf1** [comment]
👍

**Pull request merged**